CollectorUtils/UpdateGNSSSMetadata.py

```diff
@@ -0,0 +1,293 @@
+                            'fields']) + initialFeatureLayerFieldsCount) > initialFeatureLayerFieldsCount:
+                    operations.append('addToDefinition')
+            else:
+                operations.append('deleteFromDefinition')
+                gnssMetadataFields = {
+                    'fields': [{'name': 'ESRIGNSS_FIXDATETIME'},{'name': 'ESRIGNSS_RECEIVER'}, {'name': 'ESRIGNSS_H_RMS'}, {'name': 'ESRIGNSS_V_RMS'},
+                               {'name': 'ESRIGNSS_LATITUDE'}, \
+                               {'name': 'ESRIGNSS_LONGITUDE'}, {'name': 'ESRIGNSS_ALTITUDE'}, {'name': 'ESRIGNSS_PDOP'},
+                               {'name': 'ESRIGNSS_HDOP'}, \
+                               {'name': 'ESRIGNSS_VDOP'}, {'name': 'ESRIGNSS_CORRECTIONAGE'},
+                               {'name': 'ESRIGNSS_FIXTYPE'}, {'name': 'ESRIGNSS_STATIONID'}, \
+                               {'name': 'ESRIGNSS_NUMSATS'}]}
+
+            # Get index of the substring /services/ to construct admin url.
+            servicesKeywordIndex = featureServiceUrl.index('/services/')
+
+            for operation in operations:
+                # Construct admin url to Update/Add/Delete service definition
+                adminUrl = featureServiceUrl[0:servicesKeywordIndex] + '/admin' + featureServiceUrl[
+                                                                                  servicesKeywordIndex:] + '/' + operation + '?'
+
+                # Update/Add/Delete Service Definition
+                payload = ''
+                if operation == 'addToDefinition' or operation == 'deleteFromDefinition':
+                    payload = bytes(urllib.parse.urlencode({'token': token, 'f': 'json',
+                                                            operation: json.dumps(gnssMetadataFields).encode('utf8',
+                                                                                                             errors='replace')}),
+                                    'utf-8')
+                else:
+                    featureLayerJson['editingInfo'] = {'lastEditDate': ''}
+                    payload = bytes(urllib.parse.urlencode({'token': token, 'f': 'json',
+                                                            operation: json.dumps(featureLayerJson).encode('utf8',
+                                                                                                           errors='replace')}),
+                                    'utf-8')
+
+                serviceDefinitionResponse = urllib.request.urlopen(adminUrl, payload).read()
+                result = json.loads(serviceDefinitionResponse)['success']
+
+                if not result:
+                    print('Failed to update Feature layer service definition..')
+                else:
+                    print('Service definition updated successfully..')
+
+    except Exception as e:
+        print('Failed to Update Feature layer service definition')
```

Comment:
Can you print the exception `e` here as well, so we get more details of why it failed.